### PR TITLE
Fix Hudi Cert Key issue External Clients

### PR DESCRIPTION
**Pull request by moritzmeister**
JAVA NOT TESTED YET!

### CODE REVIEW

File: java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksExternalClient.java [hunk 1]
```diff
@@ -63,13 +65,26 @@
   private static final String PARAM_NAME_SECRET_STORE = "hopsworks/role/";
   private static final String PARAM_NAME_PARAMETER_STORE = "/hopsworks/role/";
 
+  private static final String MATERIAL_PASSWD = "material_passwd";
+  private static final String T_CERTIFICATE = "t_certificate";
+  private static final String K_CERTIFICATE = "k_certificate";
```
Comment (SirOibaf) [inline java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksExternalClient.java:70]: What are these for?

File: java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksHttpClient.java [hunk 2]
```diff
@@ -19,10 +19,34 @@
...
+      int content;
+      while ((content = fis.read()) != -1) {
+        sb.append((char) content);
+      }
+      return sb.toString();
+    } catch (IOException ex) {
+      LOGGER.warn("Failed to get cert password.", ex);
+    }
+    return null;
+  }
```
Comment (SirOibaf) [inline java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksHttpClient.java:51]: Use the commons-io library, there is a FileUtils.readFileToString method which makes reading a String from a file a one-liner.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksExternalClient.java (hunk 1 above)
Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksExternalClient.java]: copy paste, removed them

File: java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksHttpClient.java (hunk 2 above)
Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksHttpClient.java]: replaced as you said, dunno where I got this monster from

**Review by SirOibaf — APPROVED** [review]

**Comment by SirOibaf** [comment]
Approved

**Pull request merged**